### DISCUSSION TITLE
Using PC with Spring Cloud Stream / Kafka Binder

### DISCUSSION THREAD

**Discussion by wlaw868** [confluentinc/parallel-consumer#477] [General] [answered]
Has anyone used this PC with Kafka binder? Kindly share your experience and how easy is it?

**Comment by rkolesnev** [comment, marked as answer]
Hi,

I haven't tested the integration yet. 
But from my previous experience and understanding of how Spring Cloud Stream and Kafka Binder works - I don't think it will work with Parallel consumer (either at all or with major pitfalls).

Basically - Spring Cloud Stream handles handoff between consuming and processing of messages using binders and Spring Kafka library. 
Parallel Consumer on some level does the same - it is managing polling, hand off to processing function, committing etc.

What benefit Spring Cloud Stream would give versus either Spring Kafka or just Spring Boot application with Parallel Consumer?
I am curious what specific features you are looking to get from that combination?

**Reply by wlaw868** [reply]
Thanks for getting back to me. We are using SCS in a few projects and hoping we can stay with single Kafka interface. SCS offers functions such as record transformation via SCF, also it integrates with Cloudevents. These can be done with native Kafka consumer as well. We shall evaluate PC further and probably raise more questions here. Thanks!
